libpromises/evalfunction.c

```diff
@@ -4867,16 +4868,39 @@ static FnCallResult FnCallIPRange(EvalContext *ctx, ARG_UNUSED const Policy *pol
         return FnFailure();
     }
 
-    for (const Item *ip = EvalContextGetIpAddresses(ctx); ip != NULL; ip = ip->next)
+    // we match on VIPADDRESS iff no interfaces or IPs were requested
+    if (NULL == finalargs->next)
     {
-        Rlist *ifaces = finalargs->next;
-        // we match on VIPADDRESS iff no interfaces were requested
-        if (FuzzySetMatch(range, VIPADDRESS) == 0 && NULL == ifaces)
+        if (FuzzySetMatch(range, VIPADDRESS) == 0)
+        {
+            Log(LOG_LEVEL_DEBUG, "%s: found range %s on hostip %s, no interface", fp->name, range, VIPADDRESS);
+            return FnReturnContextTrue();
+        }
+
+        return FnReturnContextFalse();
+    }
+
+    // first try to match on the given arguments as a list of IPs
```

Comment:
It would be nice if the comments were briefly but clearly indicating the 3 different modes. For example:

```
/* Case 1: iprange(arg1) */
[...]
/* Case 2: iprange(arg1, arg2) */
[...]
/* Case 3: isipinsubnet(arg1, arg2) */
[...]
```